src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -303,14 +300,23 @@ export class OperationMetricsCollector {
         {
           this.handlers.forEach(metricsHandler => {
             if (metricsHandler.onOperationComplete) {
+              const firstResponseLatencyExpression =
+                this.methodName === MethodName.READ_ROWS ||
+                this.methodName === MethodName.READ_ROW
+                  ? {
+                      firstResponseLatency:
+                        this.firstResponseLatency ?? undefined,
+                    }
+                  : {};
```

Comment:
I think this check would be better [here, in onOperationComplete](https://github.com/googleapis/nodejs-bigtable/blob/bcb4cba1808064df11107ae9fc51c29cbbfee53a/src/client-side-metrics/gcp-metrics-handler.ts#L254).

If we put the validation here, we now have three meaningful states to keep track of for firstRepsonseLatency: float, undefined, or unset. And that opens us up to a bunch of weird edge cases if the type doesn't match what we expect when it comes time to record the metric.

All that type complexity goes away if we only record the metric for readRows, and ignore the field altogether for other rpcs

Reply:
This is a bit of an aside, but why do we use undefined here instead of null? It seems a bit confusing to me that this.firstResponseLatency is float or null, but data.firstResponseLatency is float or undefined. We wouldn't need that extra ?? operator if they used the same representation for an unset value